src/Client.ts

```diff
@@ -45,7 +45,7 @@ export class Client {
 	public static init(options: Options): Client {
 		const clientOptions: ClientOptions = {};
 		for (const i in options) {
-			if (options.hasOwnProperty(i)) {
+			if (Object.prototype.hasOwnProperty.call(options, i)) {
```

Comment:
I'm not sure which rule triggered that change but we should probably tweak the settings. This makes the code convoluted. My understanding is that it's done because JS doesn't guard against using the "hasOwnProperty" property name and this prevents using a local definition. But we shouldn't need that. https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/hasOwnProperty#using_hasownproperty_as_a_property_name

Reply:
This rule is recommended because of possible vulnerabilities -  
https://eslint.org/docs/rules/no-prototype-builtins
https://www.nickang.com/2018-02-04-developers-use-object-prototype-hasownproperty-call/

The code looks verbose but avoids the certain vulnerabilities.

Reply:
I understand the argument of the method being defined locally to something else. Or the one about creating objects without prototypes. As long as we use hasOwnProperty on objects we created, and as long as we don't do either of those, we should be good on that front.
I believe the Map attack argument is a moot point now as we ship to es2015/es6 and that's been fixed.
But better be safe than sorry. Thanks for the context.